v0/destinations/google_cloud_function/transform.js

```diff
@@ -0,0 +1,156 @@
+      };
+  }
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.body.JSON = message;
+  response.endpoint = url;
+
+  return response;
+}
+
+// Returns a batched response list for a for list of inputs(successRespList)
+function batchEvents(eventsChunk, destination) {
+  const batchedResponseList = [];
+  const {
+      enableBatchInput
+  } = destination.Config;
+  if (enableBatchInput) {
+      eventsChunk.forEach(chunk => {
+          const batchEventResponse = generateBatchedPayloadForArray(chunk);
+          batchedResponseList.push(
+              getSuccessRespEvents(
+                  batchEventResponse.batchedRequest,
+                  batchEventResponse.metadata,
+                  batchEventResponse.destination,
+                  true
+              )
+          );
+      });
+  } else {
+      eventsChunk.forEach(event => {
+          batchedResponseList.push(
+              getSuccessRespEvents(
+                  event.message,
+                  [event.metadata],
+                  destination
+              )
+          );
+      });
+
+
+  }
+  return batchedResponseList;
+}
+
+function getEventChunks(event, eventsChunk) {
+  eventsChunk.push(event);
+}
+
+const processRouterDest = async inputs => {
+  if (!Array.isArray(inputs) || inputs.length <= 0) {
+      const respEvents = getErrorRespEvents(null, 400, "Invalid event array");
+      return [respEvents];
+  }
+
+  const eventsChunk = []; // temporary variable to divide payload into chunks
+  const errorRespList = [];
+  await Promise.all(
+      inputs.map(event => {
+          try {
+              if (event.message.statusCode) {
+                  // already transformed event
+                  getEventChunks(event, eventsChunk);
+              } else {
+                  // if not transformed
+                  let response = process(event);
+                  response = Array.isArray(response) ? response : [response];
+                  response.forEach(res => {
+                      getEventChunks({
+                              message: res,
+                              metadata: event.metadata,
+                              destination: event.destination
+                          },
+                          eventsChunk
+                      );
+                  });
+              }
+          } catch (error) {
+              errorRespList.push(
+                  getErrorRespEvents(
+                      [event.metadata],
+                      error.response ? error.response.status : 400,
+                      error.message || "Error occurred while processing payload."
+                  )
+              );
+          }
+      })
+  );
+
+  const {
+      destination
+  } = inputs[0];
+  if (!destination.Config) {
+      const respEvents = getErrorRespEvents(
+          batchMetadata,
+          400,
+          "destination.Config cannot be undefined"
+      );
+      return [respEvents];
+  }
```

Comment:
We are already checking for config fields earlier. So this will not make sense here. Please remove it.